src/test/java/service/ReportGeneratorTest.java

```diff
@@ -0,0 +1,26 @@
+package service;
+
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReportGeneratorTest {
+    private static final String HEADING = "fruit,quantity";
+    private static ReportGenerator reportGenerator;
+
+    @BeforeAll
+    static void setUp() {
+        reportGenerator = new ReportGeneratorImpl();
+    }
+
+    @Test
+    void generate_validReport_ok() {
+        Map<String, Integer> data = Map.of("test1", 10,"test2",20);
```

Comment:
```suggestion
        Map<String, Integer> data = Map.of("test1", 10, "test2", 20);
```
don't forget to auto format your files before committing with Ctrl Alt L